extensions/git/src/commands.ts

```diff
@@ -3408,16 +3410,27 @@ export class CommandCenter {
 				});
 				const processors = [new RefProcessor(RefType.Head, BranchItem)];
 				const itemsProcessor = new RefItemsProcessor(repository, processors);
-				return itemsProcessor.processRefs(refs);
+				const branchItems = itemsProcessor.processRefs(refs);
+				return [createBranch, ...branchItems];
 			};
 
 			const placeHolder = l10n.t('Select a branch to create the new worktree from');
 			const choice = await this.pickRef(getBranchPicks(), placeHolder);
 
-			if (!(choice instanceof BranchItem) || !choice.refName) {
+			if (choice === createBranch) {
+				const branchName = await this.promptForBranchName(repository);
+
+				if (!branchName) {
+					return;
+				}
+
+				await repository.branch(branchName, false, 'HEAD');
```

Comment:
Instead of creating the branch, let's use the `-b` option when creating a worktree as that will create a new worktree for a new branch in one go - https://git-scm.com/docs/git-worktree#Documentation/git-worktree.txt--bnew-branch